src/transformers/utils/quantization_config.py

```diff
@@ -344,11 +344,13 @@ class GPTQConfig(QuantizationConfigMixin):
             The batch size used when processing the dataset
         pad_token_id (`int`, *optional*):
             The pad token id. Needed to prepare the dataset when `batch_size` > 1.
-        disable_exllama (`bool`, *optional*, defaults to `False`):
+        use_exllama (`bool`, *optional*, defaults to `True`):
             Whether to use exllama backend. Only works with `bits` = 4.
         max_input_length (`int`, *optional*):
             The maximum input length. This is needed to initialize a buffer that depends on the maximum expected input
             length. It is specific to the exllama backend with act-order.
+        use_exllama_v2 (`bool`, *optional*, defaults to `False`):
+            Whether to use exllamav2 backend. Only works with `bits` = 4.
```

Comment:
Having both of these arguments isn't a great pattern - we don't want to have to add a new config argument and then make sure w'ere correctly handling the compatibility of all of their settings e.g. here we should raise an exception if both `use_exllama` and `use_exllama_v2` are `True`, which quickly becomes hard to manage with more options. 

How about having an exllama enum or a dictionary specifying exllama configuriation e.g `exllama_config={"version": 2}`?

Reply:
Yeah, let's do that. It will be easier to manage the options.